src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SubSourceLogic.cs

```diff
@@ -78,7 +125,9 @@ private class CloseRevokedPartitions { }
         /// We have created a source for these partitions, but it has not started up and is not in subSources yet.
         /// </summary>
         private IImmutableSet<TopicPartition> _partitionsInStartup = ImmutableHashSet<TopicPartition>.Empty;
-        private IImmutableDictionary<TopicPartition, IControl> _subSources = ImmutableDictionary<TopicPartition, IControl>.Empty;
+
+        private IImmutableDictionary<TopicPartition, SubSourceStageLogicControl> _subSources =
```

Comment:
We now store the `SubSourceStageLogicControl` type we defined earlier and associate it with its given `TopicPartition` - remember the `SubSourceLogic` is the parent object that emits the sub-streams, so this stage has to track and manage the lifecycle of all of them.